Source/ablastr/utils/SignalHandling.cpp

```diff
@@ -0,0 +1,180 @@
+/* Copyright 2022 Philip Miller
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "SignalHandling.H"
+
+#include <AMReX.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_IParser.H>
+
+#include <cctype>
+
+// For sigaction() et al.
+#if defined(__linux__) || defined(__APPLE__)
+#include <signal.h>
+#endif
+
+std::atomic<bool> SignalState::signal_received_flags[NUM_SIGNALS];
+bool SignalState::signal_conf_requests_break[NUM_SIGNALS];
+bool SignalState::signal_conf_requests_checkpoint[NUM_SIGNALS];
```

Comment:
If/when more different actions are added, it would maybe make sense to turn these into parts of a 2D array `[SIGNAL_REQUESTS_MAX][NUM_SIGNALS]`